stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/options/MessageOptionsDialogFragment.kt

```diff
@@ -252,14 +253,14 @@ internal class MessageOptionsDialogFragment : FullScreenDialogFragment() {
     }
 
     internal class MessageOptionsHandlers(
-        val threadReplyHandler: (Message) -> Unit,
-        val retryHandler: (Message) -> Unit,
-        val editClickHandler: (Message) -> Unit,
-        val flagClickHandler: (Message) -> Unit,
-        val muteClickHandler: (User) -> Unit,
-        val blockClickHandler: (User) -> Unit,
-        val deleteClickHandler: (Message) -> Unit,
-        val replyClickHandler: (String, Message) -> Unit,
+        val threadReplyHandler: MessageListView.StartThreadHandler,
+        val retryHandler: MessageListView.MessageRetryHandler,
+        val editClickHandler: MessageListView.MessageEditHandler,
+        val flagClickHandler: MessageListView.MessageFlagHandler,
+        val muteClickHandler: MessageListView.MuteUserHandler,
+        val blockClickHandler: (user: User) -> Unit,
```

Comment:
This is really awkward like this, but we can't take the `Handler` type defined in `MessageListView` as the `Channel` parameter for its method is being filled in by `MessageListView`, and not available here. However, this is all internal API, so we can refactor it later.

Reply:
I think that we don't need the whole `Channel` object to block the user. It should be enough to pass `Channel::cid`

Reply:
Refactored so that we pass in the channel ID instead - take a look